cfme/tests/automate/test_service_automate.py

```diff
@@ -455,3 +459,184 @@ def test_retire_vm_now(setup_provider, full_template_vm, new_user):
             full_template_vm.retire()
             assert full_template_vm.wait_for_vm_state_change(desired_state="retired", timeout=720,
                                                              from_details=True)
+
+
+@pytest.fixture
+def custom_prov_data(request, provisioning):
+    prov_data = {
+        "catalog": {'vm_name': random_name(prefix="test_")},
+        "environment": {'automatic_placement': True},
+        "network": {'vlan': partial_match(provisioning['vlan'])}
+    }
+    prov_data.update(request.param)
+    prov_data['catalog']['catalog_name'] = {'name': provisioning["template"]}
+    return prov_data
+
+
+@pytest.fixture(scope='function')
+def domain_setup(domain):
+    """This fixture used to setup the domain structure"""
+    # Instantiating class 'QuotaMethods'
+    klass = (
+        domain.parent.instantiate(name="ManageIQ")
+        .namespaces.instantiate(name="System")
+        .namespaces.instantiate(name="CommonMethods")
+        .classes.instantiate(name="QuotaMethods")
+    )
+
+    # Instantiating method 'requested' and copying it to new domain
+    klass.methods.instantiate(name="requested").copy_to(domain.name)
+
+    method = (
+        domain.parent.instantiate(name=f"{domain.name}")
+        .namespaces.instantiate(name="System")
+        .namespaces.instantiate(name="CommonMethods")
+        .classes.instantiate(name="QuotaMethods")
+        .methods.instantiate(name="requested")
+    )
+    yield
+    method.delete_if_exists()
+
+
+@pytest.fixture(scope='module')
+def tenants_setup(appliance):
+    """This fixture creates two parent tenants"""
+    parent_tenant = appliance.collections.tenants.create(
+        name=fauxfactory.gen_alphanumeric(18, start="test_parent_"),
+        description=fauxfactory.gen_alphanumeric(18, start="parent_desc_"),
+        parent=appliance.collections.tenants.get_root_tenant()
+    )
+    child_tenant = appliance.collections.tenants.create(
+        name=fauxfactory.gen_alphanumeric(18, start="test_parent_"),
+        description=fauxfactory.gen_alphanumeric(18, start="parent_desc_"),
+        parent=parent_tenant
+    )
+    return parent_tenant, child_tenant
+
+
+@pytest.fixture
+def set_child_tenant_quota(request, appliance, tenants_setup):
+    parent_tenant, child_tenant = tenants_setup
+    field_value = request.param
+    tenant_quota_data = {}
+    for field, value in field_value:
+        tenant_quota_data.update({f"{field}_cb": True, field: value})
+    child_tenant.set_quota(**tenant_quota_data)
+    yield
+    appliance.server.login_admin()
```

Comment:
Please add a comment as to why logging again is required here

Reply:
No need to login again. Hence removed.